frontends/api/src/hooks/learningResources/index.ts

```diff
@@ -261,8 +262,8 @@ const useUserListDestroy = () => {
   return useMutation({
     mutationFn: (params: ULDestroyRequest) =>
       userListsApi.userlistsDestroy(params),
-    onSettled: (_data, _err, vars) => {
-      invalidateResourceQueries(queryClient, vars.id)
+    onSettled: (_data, _err) => {
+      queryClient.invalidateQueries(learningResources.userlists._ctx.list._def)
```

Comment:
There's a small bug here that you can experience in UI as follows:
1. Navigate to dashboard userlists
2. Create a userlist
3. navigate to dashboard "home"
4. bookmark a currently unbookmarked resource into your new list
5. navigate back to userlists
6. delete the userlist
7. navigate back to "home" ... you'll see the resource is still bookmarked.

Instead:
```suggestion
    onSettled: (_data, _err, vars) => {
        useUserListUpdate(vars.id)
```
This will invalidate the userlist and any resources that are using that userlist (otherwise their bookmarks will still show filled).

I will say...
- all this invalidation stuff is complicated. Part of me just wants to blow away the whole cache, but that doesn't seem great for some things, like sorting resources.
- we don't have a great way to test this invalidation stuff right now. The tests in `index.test.ts` are....not ideal. (neither is the filename)